core/modules/filter/filter.pages.inc

```diff
@@ -100,6 +100,25 @@ function _filter_link_autocomplete($string) {
       $matches['/' . $path] = check_plain($result->title) . ' (' . t('Content') . ')';
       $match_count++;
     }
+    
+    // Get file matches.
+    $results = db_select('file_managed')
+      ->fields('file_managed', array('fid', 'filename', 'uri', 'filemime', 'status'))
+      ->condition('filename', db_like($string) . '%', 'LIKE')
+      ->condition('status', 1)
+      ->condition('filemime', 'image%', 'NOT LIKE')
+      ->orderBy('fid', 'DESC')
+      ->range(0, 10 - $match_count)
+      ->execute();
+    foreach ($results as $result) {
+      // Try to make a local path if possible for better portability.
+      $absolute_path = parse_url($GLOBALS['base_url'], PHP_URL_PATH) . '/';
+      $url = file_create_url($result->uri);
+      $path = str_replace($GLOBALS['base_url'] . '/', $absolute_path, $url);
```

Comment:
I don't really understand what you try to achieve in the last 3 lines. Also, the usage of `base_path()` can make the code cleaner.  But why the output of `file_create_url()` is not good as is ?

Reply:
This is to do with the problem of Backdrop installed in subdirectories identified and solved elsewhere in this file thanks to #2068 which was for images but also applies to files uploaded via WYSIWYG.

Reply:
this is why `base_path()` exists, isn't it ?

Reply:
Well, base_path() certainly works for my installations in subdirectories, and I am not sure why issue #2562 has been fixed this other way. I have copied that solution thinking there must be configurations where it is needed rather than base_path().

Reply:
it could be that the other solution was imperfect. I definitely think `base_path` is the way to go here. Can we try it with base_path() instead? I hate to see those globals used this way. We can file a follow-up to https://github.com/backdrop/backdrop/pull/2068 to fix that issue.